Dockerfile.tool-router.optimized

```
@@ -0,0 +1,61 @@
+# syntax=docker/dockerfile:1.6
+# Optimized multi-stage build for MCP Gateway Tool Router
+
+# Stage 1: Dependencies with BuildKit cache mount
+FROM python:3.12-alpine AS deps
+WORKDIR /app
+
+# Copy only dependency files first (better layer caching)
+COPY requirements.txt .
+
+# Use BuildKit cache mount for pip cache (dramatic speed improvement)
+RUN --mount=type=cache,target=/root/.cache/pip \
```

Hadolint (2.14.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)